File: app/src/main/java/xyz/sleekstats/completist/view/WelcomeActivity.java
package xyz.sleekstats.completist.view;

import android.content.Intent;
import android.net.Uri;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.google.ads.consent.ConsentForm;
import com.google.ads.consent.ConsentFormListener;
import com.google.ads.consent.ConsentInfoUpdateListener;
import com.google.ads.consent.ConsentInformation;
import com.google.ads.consent.ConsentStatus;
import com.google.ads.consent.DebugGeography;
import com.google.android.gms.ads.MobileAds;

import java.net.MalformedURLException;
import java.net.URL;

import xyz.sleekstats.completist.R;

public class WelcomeActivity extends AppCompatActivity {

    private ConsentForm form;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);
        MobileAds.initialize(this, "ca-app-pub-5443559095909539~1441728975");

//        checkForConsent();
    }

    public void goToMain(View view) {
        view.setClickable(false);
        Intent intent;
        if(view.getId() == R.id.tmdb_acknowledgment_imageView) {
            view.setClickable(true);
            intent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://www.themoviedb.org/"));
            startActivity(intent);
        } else {
            intent = new Intent(WelcomeActivity.this, MainActivity.class);
            startActivity(intent);
            finish();
        }
    }

    private void checkForConsent() {
        Log.d("pcpo", "checkForConsent");
        ConsentInformation consentInformation = ConsentInformation.getInstance(WelcomeActivity.this);
        consentInformation.setDebugGeography(DebugGeography.DEBUG_GEOGRAPHY_EEA);
        String[] publisherIds = {"pub-5443559095909539"};

        consentInformation.requestConsentInfoUpdate(publisherIds, new ConsentInfoUpdateListener() {
            @Override
            public void onConsentInfoUpdated(ConsentStatus consentStatus) {
                switch (consentStatus) {
                    case UNKNOWN:
                        if (ConsentInformation.getInstance(getBaseContext())
                                .isRequestLocationInEeaOrUnknown()) {
                            requestConsent();
                        }
                }
            }
            @Override
            public void onFailedToUpdateConsentInfo(String errorDescription) {
            }
        });
    }

    private void requestConsent() {
        Log.d("pcpo", "requestConsent");
        URL privacyUrl = null;
        try {
            privacyUrl = new URL("https://sites.google.com/site/sleekstatsprivacypolicy/");
        } catch (MalformedURLException e) {
            Log.d("pcpo", "MalformedURLException " + e.getMessage());
            e.printStackTrace();
        }
        form = new ConsentForm.Builder(WelcomeActivity.this, privacyUrl)
                .withListener(new ConsentFormListener() {
                    @Override
                    public void onConsentFormLoaded() {
                        Log.d("pcpo", "onConsentFormLoaded");
                        if (form != null) {
                            form.show();
                            Log.d("pcpo", "form.show");
                        }
                    }

                    @Override
                    public void onConsentFormError(String reason) {
                        super.onConsentFormError(reason);
                        Log.d("pcpo", "onConsentFormError  " + reason);
                    }

                    @Override
                    public void onConsentFormOpened() {
                        super.onConsentFormOpened();
                        Log.d("pcpo", "onConsentFormOpened");
                    }

                    @Override
                    public void onConsentFormClosed(ConsentStatus consentStatus, Boolean userPrefersAdFree) {
                        super.onConsentFormClosed(consentStatus, userPrefersAdFree);
                        Log.d("pcpo", "onConsentFormClosed");
                    }
                })
                .withAdFreeOption()
                .build();
        form.load();
        Log.d("pcpo", "form.load");
    }
}

File: app/src/main/java/xyz/sleekstats/completist/view/WelcomeActivity.kt
package xyz.sleekstats.completist.view

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.google.ads.consent.*
import com.google.android.gms.ads.MobileAds
import xyz.sleekstats.completist.R
import xyz.sleekstats.completist.view.MainActivity
import java.net.MalformedURLException
import java.net.URL

class WelcomeActivity : AppCompatActivity() {
    private var form: ConsentForm? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_welcome)
        MobileAds.initialize(this, "ca-app-pub-5443559095909539~1441728975")

//        checkForConsent();
    }

    fun goToMain(view: View) {
        view.isClickable = false
        val intent: Intent
        if (view.id == R.id.tmdb_acknowledgment_imageView) {
            view.isClickable = true
            intent = Intent(Intent.ACTION_VIEW, Uri.parse("https://www.themoviedb.org/"))
            startActivity(intent)
        } else {
            intent = Intent(this@WelcomeActivity, MainActivity::class.java)
            startActivity(intent)
            finish()
        }
    }

    private fun checkForConsent() {
        val publisherIds = arrayOf("pub-5443559095909539")
        ConsentInformation.getInstance(this@WelcomeActivity).apply {
            debugGeography = DebugGeography.DEBUG_GEOGRAPHY_EEA
            requestConsentInfoUpdate(publisherIds, object : ConsentInfoUpdateListener {
                override fun onConsentInfoUpdated(consentStatus: ConsentStatus) {
                    if (consentStatus == ConsentStatus.UNKNOWN && ConsentInformation.getInstance(baseContext).isRequestLocationInEeaOrUnknown) {
                        requestConsent()
                    }
                }

                override fun onFailedToUpdateConsentInfo(errorDescription: String) {}
            })
        }
    }

    private fun requestConsent() {
        var privacyUrl: URL? = null
        try {
            privacyUrl = URL("https://sites.google.com/site/sleekstatsprivacypolicy/")
        } catch (e: MalformedURLException) {
            Log.d("pcpo", "MalformedURLException " + e.message)
            e.printStackTrace()
        }
        form = ConsentForm.Builder(this@WelcomeActivity, privacyUrl)
                .withListener(object : ConsentFormListener() {
                    override fun onConsentFormLoaded() {
                        if (form != null) {
                            form!!.show()
                        }
                    }
                })
                .withAdFreeOption()
                .build()
        form?.load()
    }
}